server/src/controllers/Venue/resolver.ts

```diff
@@ -102,6 +104,100 @@ export class VenueResolver {
     @Arg('_onlyUsedForAuth', () => Int) _onlyUsedForAuth: number,
   ): Promise<{ id: number }> {
     // TODO: handle deletion of non-existent venue
+    const users = await prisma.users.findMany({
+      where: {
+        user_events: {
+          some: {
+            event: {
+              canceled: false,
+              ends_at: { gt: new Date() },
+              venue_id: id,
+            },
+            subscribed: true,
+            rsvp: { name: 'yes' },
+          },
+        },
+      },
+      include: {
+        user_events: {
+          where: {
+            event: {
+              canceled: false,
+              ends_at: { gt: new Date() },
+              venue_id: id,
+            },
+            subscribed: true,
+            rsvp: { name: 'yes' },
+          },
+        },
+      },
+    });
+
+    const venue = await prisma.venues.findUniqueOrThrow({
+      where: { id: id },
+    });
+
+    const chapterId = venue.chapter_id;
+
+    const unsubscribeOptions = (
+      currentUserId: number,
+      currentEventId: number,
+    ) =>
+      eventUnsubscribeOptions({
+        chapterId: chapterId,
+        eventId: currentEventId,
+        userId: currentUserId,
+      });
+
+    const eventList = (event: events[], currentUserId: number) => {
+      event.map(
+        ({ name, id }) => `
+      <tr>
+      <td>${name}</td>
+      <td>${unsubscribeOptions(id, currentUserId)}</td>
+     </tr>
+  `,
+      );
+    };
+    const emailSubject = `Events hosted at ${venue.name} won't be hosted there anymore`;
+    const emailContent = (
+      event: events[],
+      currentUserId: number,
+    ) => `The events related to ${
+      venue.name
+    } won't be host locally anymore.<br />
+    <table>
+    <thead>
+        <tr>
+          <th>Event</th>
+          <th>Cancel RSVP</th>
+        </tr>
+    </thead>
+    <tbody>
+       ${eventList(event, currentUserId)}
+    </tbody>
+</table>`;
+
+    for (const { email, id: currentUserId } of users) {
+      const events = await prisma.events.findMany({
+        where: {
+          canceled: false,
+          ends_at: { gt: new Date() },
+          venue_id: id,
+          event_users: { every: { user_id: currentUserId, subscribed: true } },
```

Comment:
Random question, do we need `subscribed` and `rsvp` here? We have already found user_id, so it should select the same user, and we won't need to filter the user again

```suggestion
          event_users: { every: { user_id: currentUserId } },
```

Reply:
Good news is additional queries are not needed, because `event` has relation with `user_event`, so every `user_event` can already link to the affected event. Although one more line is needed to include the actual event in the other query :eyes:

```typescript
    const users = await prisma.users.findMany({
      where: {
        user_events: {
          some: {
            event: {
              canceled: false,
              ends_at: { gt: new Date() },
              venue_id: id,
            },
            subscribed: true,
            rsvp: { name: 'yes' },
          },
        },
      },
      include: {
        user_events: {
          include: { event: true },
          where: {
            event: {
              canceled: false,
              ends_at: { gt: new Date() },
              venue_id: id,
            },
            subscribed: true,
            rsvp: { name: 'yes' },
          },
        },
      },
    });
```